pkg/controller/daemon/daemoncontroller.go

```diff
@@ -825,21 +991,28 @@ func (dsc *DaemonSetsController) syncDaemonSet(key string) error {
 		return dsc.updateDaemonSetStatus(ds)
 	}
 
-	if err := dsc.manage(ds); err != nil {
+	hash, err := dsc.manage(ds)
+	if err != nil {
 		return err
 	}
 
 	// Process rolling updates if we're ready.
 	if dsc.expectations.SatisfiedExpectations(dsKey) {
 		switch ds.Spec.UpdateStrategy.Type {
+		case extensions.OnDeleteDaemonSetStrategyType:
 		case extensions.RollingUpdateDaemonSetStrategyType:
-			err = dsc.rollingUpdate(ds)
+			err = dsc.rollingUpdate(ds, hash)
 		}
 		if err != nil {
 			return err
 		}
 	}
 
+	err = dsc.cleanupHistory(ds)
```

Comment:
Yes, `rollingUpdate` calls `syncNodes` which creates & deletes pods and waits for those operations to complete